From 0ca56b6450d171d166d8a886ee493ebb6583daa1 Mon Sep 17 00:00:00 2001
From: Du, Changbin <changbinx.du@intel.com>
Date: Thu, 20 Jun 2013 01:39:55 +0800
Subject: remove dirty file if readObject fails

BZ: 116480

Remove dirty file if readObject fails to read all object data. Else
broken files will leave in filesystem.

Category: AOSP Improvement
Domain: USB MTP-PTP
Origin: Internal
Upstream-Candidate: No, proprietary

Change-Id: Ic18c3d3e0abc62ed1871268afd519502d7499111
Orig-Change-Id: I763523a55fb2a049c2b0ff68acc528b21c92ca85
Signed-off-by: Du, Changbin <changbinx.du@intel.com>
---
 media/mtp/MtpDevice.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)
 mode change 100644 => 100755 media/mtp/MtpDevice.cpp

diff --git a/media/mtp/MtpDevice.cpp b/media/mtp/MtpDevice.cpp
old mode 100644
new mode 100755
index 603b039..489620b
--- a/media/mtp/MtpDevice.cpp
+++ b/media/mtp/MtpDevice.cpp
@@ -778,6 +778,8 @@ bool MtpDevice::readObject(MtpObjectHandle handle, const char* destPath, int gro
 
 fail:
     ::close(fd);
+    if (!result)
+        unlink(destPath);
     return result;
 }
 
-- 
1.7.4.1

